conans/test/integration/layout/test_layout_generate.py

```diff
@@ -48,3 +49,50 @@ def build(self):
     assert os.path.exists(os.path.join(c.current_folder, "mybuild", "include", "myfile.txt"))
 
 
+def test_generate_source_folder():
+    c = TestClient()
+    conanfile = textwrap.dedent("""
+        import os
+        from conan import ConanFile
+        from conan.tools.cmake import cmake_layout
+
+        class Pkg(ConanFile):
+            settings = "os", "compiler", "build_type", "arch"
+
+            def layout(self):
+                cmake_layout(self)
+
+            def generate(self):
+                self.output.info("PKG_CONFIG_PATH {}!".format(os.path.exists(self.source_folder)))
```

Comment:
how is related the PKG_CONFIG_PATH with this change in the code base?

Reply:
It is just an example, the original reported issue wanted to set-up the PKG_CONFIG_PATH, but it could be any other message.